File: Countries/js/2.js
// What countries both begin and end with a vowel?

let main = () => {
    
    const fs = require("fs");
    let buffer_arry = fs.readFileSync("Countries/countries.txt", "utf-8");
    let in_file = buffer_arry.split("\n");
    let vowels = ["A", "E", "I", "O", "U"];

    // Function checks fist char of word for vowel
    let check_vowels_first = (word) => {

        for(let i = 0; i < vowels.length; i++) {
            if(word[0] == vowels[i]) {
                for(let j = 0; j < vowels.length; j++) {
                    if(word[word.length - 1] == vowels[j]) {
                        return true
                    };
                };
            };
        
        };
        return false;
    };

    // Main function calls
    for(let i = 0; i < in_file.length; i++) {
        if(check_vowels_first(in_file[i].toUpperCase())) {
            console.log(in_file[i]);
        };
    };
};

main();

File: Countries/python/2.py
# What countries both begin and end with a vowel?
def main():

    in_file = open("../countries.txt", "r")

    be_vowel_list = ["a", "e", "i", "o", "u"]

    def check_be_vowel(word, l_index):
        for i in range(len(be_vowel_list)):
            if(word[0].lower() == be_vowel_list[i]):
                for j in range(len(be_vowel_list)):
                    if(word[l_index].lower() == be_vowel_list[j]):
                        print(word, end="")
                        return
                return

    for words in in_file:
        be_vowel_lindex = len(words) - 2 # index position of last char
        check_be_vowel(words, be_vowel_lindex)

if(__name__  == "__main__"):
    main()